docs/releases.md

```diff
@@ -0,0 +1,28 @@
+# Releases
+
+## Supported Versions
+
+Versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version,
+following Semantic Versioning terminology.
+
+We maintain release branches for the most recent two minor releases.
+
+Fixes may be backported to release branches, depending on severity, risk, and, feasibility.
+
+If a release contains breaking changes, or CVE fixes, this will documented in the release notes.
+
+## Supported Version Skew
+
+Both the `argo-server` and `argocli` should be the same version as the controller.
```

Comment:
The `argocli` and the `argo-server` are the same binary/image. Should we just say "`argocli` (which contains the Argo Server"?

Reply:
I think most users do not know that.